test/tests/functional/pbs_sched_fifo.py

```diff
@@ -0,0 +1,90 @@
+# coding: utf-8
+
+# Copyright (C) 1994-2018 Altair Engineering, Inc.
+# For more information, contact Altair at www.altair.com.
+#
+# This file is part of the PBS Professional ("PBS Pro") software.
+#
+# Open Source License Information:
+#
+# PBS Pro is free software. You can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free
+# Software Foundation, either version 3 of the License, or (at your option) any
+# later version.
+#
+# PBS Pro is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.
+# See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+# Commercial License Information:
+#
+# For a copy of the commercial license terms and conditions,
+# go to: (http://www.pbspro.com/UserArea/agreement.html)
+# or contact the Altair Legal Department.
+#
+# Altair’s dual-license business model allows companies, individuals, and
+# organizations to create proprietary derivative works of PBS Pro and
+# distribute them - whether embedded or bundled with other software -
+# under a commercial license agreement.
+#
+# Use of Altair’s trademarks, including but not limited to "PBS™",
+# "PBS Professional®", and "PBS Pro™" and Altair’s logos is subject to Altair's
+# trademark licensing policies.
+
+from tests.functional import *
+
+
+class TestSchedFifo(TestFunctional):
+    """
+    Test suite for PBSPro's FIFO scheduling
+    """
+    def test_sched_fifo(self):
+        """
+        Check that FIFO works.
```

Comment:
You're test logic is flawed.  First you submit your jobs in the wrong order.  You need to submit them j1=workq, j2=workq2, j3=workq.  You submit workq2, workq2, workq.  The way you do it will reproduce the the same behavior regardless of the value of by_queue.  Second, by only having 1 cpu and deleting a job, you are only really testing that the scheduler is running a job in the highest priority queue.

What you need to do is the following:
Have 2 cpus
turn scheduling off
submit jobs j1=workq, j2=workq2, j3=workq
turn scheduling on
Check that j1, and j2 are running and j3 is queued.

Before this bug, j1 and j3 would be running.

Reply:
@bhroam, I was following this logic since this is what was mentioned in the bug description. 

I tested the scenario you mentioned here, and in both cases j1 and j2 were running (and j3 was queued). I tested another scenario where I submit 3 jobs to workq2 and 1 to workq, with 2 ncpus, the first job from workq2 and the last job (on workq) were run. Check this file for reference: [Reproduction_Attempt_FIFO.txt](https://github.com/PBSPro/pbspro/files/1681049/Reproduction_Attempt_FIFO.txt)

So, maybe I can change the test case to this:-
Have 2 ncpus
Turn scheduling off
Submit jobs j1 = workq2, j2 = workq2, j3 = workq2, j4 = workq
Turn scheduling on
Check whether j1 and j2 are running and rest are queued.
Right now, without this fix: j1 and j4 would be running.